mllib/src/test/scala/org/apache/spark/ml/clustering/KMeansSuite.scala

```diff
@@ -139,16 +146,61 @@ class KMeansSuite extends SparkFunSuite with MLlibTestSparkContext with DefaultR
     val kmeans = new KMeans()
     testEstimatorAndModelReadWrite(kmeans, dataset, KMeansSuite.allParamSettings, checkModelData)
   }
+
+  test("Initialize using given cluster centers") {
+    val kmeans = new KMeans().setK(k).setSeed(1).setMaxIter(1)
+    val oneIterModel = kmeans.fit(dataset)
+    val twoIterModel = kmeans.copy(ParamMap(ParamPair(kmeans.maxIter, 2))).fit(dataset)
+    val oneMoreIterModel = kmeans.setInitialModel(oneIterModel).fit(dataset)
+
+    twoIterModel.clusterCenters.zip(oneMoreIterModel.clusterCenters)
+      .foreach { case (center1, center2) => assert(center1 ~== center2 absTol 1E-8) }
+  }
+
+  test("Initialize using wrong model") {
+    val kmeans = new KMeans().setK(k).setSeed(1).setMaxIter(10)
+    val wrongTypeModel = new KMeansSuite.MockModel()
+    assert(!kmeans.setInitialModel(wrongTypeModel).isSet(kmeans.initialModel))
+
+    val wrongKModel = KMeansSuite.generateKMeansModel(3, k + 1)
+    intercept[IllegalArgumentException] {
```

Comment:
In other places with similar checks in ML, we typically check that the error message contains the expected message. I'd prefer to keep with that pattern here.